core/imageroot/usr/local/agent/bin/redis-restore

```diff
@@ -32,4 +32,6 @@ except IndexError:
     ttl = "0"
 
 # Read raw binary data from stdin
-rdb.restore(key, ttl, sys.stdin.buffer.raw.read())
+data = sys.stdin.buffer.raw.read()
```

Comment:
redis should not complain about an empty restore

Reply:
A zero length buffer/file throws an exception. A valid value dump is a non-empty file, in any case.